src/Analyzers/CSharp/Analyzers/UseCollectionExpression/UseCollectionExpressionHelpers.cs

```diff
@@ -91,7 +91,12 @@ internal static class UseCollectionExpressionHelpers
         // Ensure that we have a collection conversion with the replacement.  If not, this wasn't a legal replacement
         // (for example, we're trying to replace an expression that is converted to something that isn't even a
         // collection type).
+        //
+        // Note: an identity conversion is always legal without needing any more checks.
         var conversion = speculationAnalyzer.SpeculativeSemanticModel.GetConversion(speculationAnalyzer.ReplacedExpression, cancellationToken);
+        if (conversion.IsIdentity)
```

Comment:
@cston  this appears to be an error.  Once this merged, i'll open a bug on this.